pkg/integrations/kafka_exporter/kafka_exporter.go

```diff
@@ -0,0 +1,168 @@
+
+	// If true, all scrapes will trigger kafka operations otherwise, they will share results. WARN: This should be disabled on large clusters
+	AllowConcurrent bool `yaml:"allow_concurrency,omitempty"`
+
+	// Maximum number of offsets to store in the interpolation table for a partition
+	MaxOffsets int `yaml:"max_offsets,omitempty"`
+
+	// How frequently should the interpolation table be pruned, in seconds
+	PruneIntervalSeconds int `yaml:"prune_interval_seconds,omitempty"`
+
+	// Regex filter for topics to be monitored
+	TopicsFilter string `yaml:"topics_filter_regex,omitempty"`
+
+	// Regex filter for consumer groups to be monitored
+	GroupFilter string `yaml:"groups_filter_regex,omitempty"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Name returns the name of the integration that this config represents.
+func (c *Config) Name() string {
+	return "kafka_exporter"
+}
+
+// CommonConfig returns the common settings shared across all configs for
+// integrations.
+func (c *Config) CommonConfig() config.Common {
+	return c.Common
+}
+
+// NewIntegration creates a new elasticsearch_exporter
+func (c *Config) NewIntegration(logger log.Logger) (integrations.Integration, error) {
+	return New(logger, c)
+}
+
+func init() {
+	integrations.RegisterIntegration(&Config{})
+}
+
+// New creates a new kafka_exporter integration.
+func New(logger log.Logger, c *Config) (integrations.Integration, error) {
+	if len(c.KafkaURIs) == 0 || c.KafkaURIs[0] == "" {
+		return nil, fmt.Errorf("empty kafka_uris provided")
+	}
+	if c.UseTLS && (c.CertFile == "" || c.KeyFile == "") {
+		return nil, fmt.Errorf("tls is enabled but key pair was not provided")
+	}
+	if c.UseSASL && (c.SASLPassword == "" || c.SASLUsername == "") {
+		return nil, fmt.Errorf("SASL is enabled but username or password was not provided")
+	}
+	if c.UseZooKeeperLag && (len(c.ZookeeperURIs) == 0 || c.ZookeeperURIs[0] == "") {
+		return nil, fmt.Errorf("zookeeper lag is enabled but no zookeeper uri was provided")
+	}
+
+	var options kafka_exporter.Options
+	options.Uri = c.KafkaURIs
+	options.UseSASL = c.UseSASL
+	options.UseSASLHandshake = c.UseSASLHandshake
+	options.SaslUsername = c.SASLUsername
+	options.SaslPassword = c.SASLPassword
+	options.SaslMechanism = c.SASLMechanism
+	options.UseTLS = c.UseTLS
+	options.TlsCAFile = c.CAFile
+	options.TlsCertFile = c.CertFile
+	options.TlsKeyFile = c.KeyFile
+	options.TlsInsecureSkipTLSVerify = c.InsecureSkipTLSVerify
+	options.KafkaVersion = c.KafkaVersion
+	options.UseZooKeeperLag = c.UseZooKeeperLag
+	options.UriZookeeper = c.ZookeeperURIs
+	options.Labels = c.ClusterName
+	options.MetadataRefreshInterval = c.MetadataRefreshInterval
+	options.AllowConcurrent = c.AllowConcurrent
+	options.MaxOffsets = c.MaxOffsets
+	options.PruneIntervalSeconds = c.PruneIntervalSeconds
```

Comment:
I'd probably just use the inline initialization here: 

```suggestion
	options := kafaka_exporter.Options{
		Uri:                      c.KafkaURIs,
		UseSASL:                  c.UseSASL,
		UseSASLHandshake:         c.UseSASLHandshake,
		SaslUsername:             c.SASLUsername,
		SaslPassword:             c.SASLPassword,
		SaslMechanism:            c.SASLMechanism,
		UseTLS:                   c.UseTLS,
		TlsCAFile:                c.CAFile,
		TlsCertFile:              c.CertFile,
		TlsKeyFile:               c.KeyFile,
		TlsInsecureSkipTLSVerify: c.InsecureSkipTLSVerify,
		KafkaVersion:             c.KafkaVersion,
		UseZooKeeperLag:          c.UseZooKeeperLag,
		UriZookeeper:             c.ZookeeperURIs,
		Labels:                   c.ClusterName,
		MetadataRefreshInterval:  c.MetadataRefreshInterval,
		AllowConcurrent:          c.AllowConcurrent,
		MaxOffsets:               c.MaxOffsets,
		PruneIntervalSeconds:     c.PruneIntervalSeconds,
	}
```